File: salt/java/src/main/java/linan/UserRepository.java
package linan;

import java.util.List;

import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.CrudRepository;
import org.springframework.data.repository.query.Param;

public interface UserRepository extends CrudRepository<User, Long> {

  User findById(Long id);

  List<User> findByAccount(String account);

  List<User> findAll();

  @Query(value="select * from user where id > :id", nativeQuery = true)
  List<User> fix(@Param("id") Long id);

//  @Query(value="select * from user where id > ?1", nativeQuery = true)
//  List<User> fix(Long id);
}

File: salt/kotlin/src/main/kotlin/salt/repository/UserRepository.kt
package salt.repository

import org.springframework.data.repository.CrudRepository
import org.springframework.transaction.annotation.Transactional

import salt.entity.UserEntity

@Transactional
interface UserRepository : CrudRepository<UserEntity, Long> {
  fun findByAccount(account: String): UserEntity?
  override fun findAll(): List<UserEntity>?
}